### PR TITLE
Fix endpoints and add rename endpoint

### PR DESCRIPTION
**Pull request by anirudhb**
Title

### CODE REVIEW

File: pkg/api/apps/routes.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ import "github.com/gin-gonic/gin"
 func SetupRoutes(r *gin.RouterGroup) {
 	r.POST("/", handlePOSTApp)
 	r.GET("/:id", handleGETApp)
+	r.POST("/:id/rename", handlePOSTRename)
```
Comment (cjdenio) [inline pkg/api/apps/routes.go]: This should be `PATCH /apps/:id`

File: pkg/api/teams/routes.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ func SetupRoutes(r *gin.RouterGroup) {
 	r.POST("/", handlePOSTTeam)
 	r.GET("/:id", handleGETTeam)
 	r.DELETE("/:id", handleDELETETeam)
+	r.POST("/:id/rename", handlePOSTRename)
```
Comment (cjdenio) [inline pkg/api/teams/routes.go]: This should be merged into the `PATCH` endpoint I added in my PR

**Review by cjdenio — CHANGES_REQUESTED** [review]

**Review by cjdenio — APPROVED** [review]
lgtm!

File: pkg/api/apps/delete.go [hunk 3]
```diff
@@ -16,9 +16,9 @@ func handleDELETEApp(c *gin.Context) {
 		return
 	}
 
-	result := db.DB.
-		Joins("INNER JOIN team_users ON team_users.team_id = apps.team_id").
-		Delete(&db.App{}, "apps.id = ? AND team_users.user_id = ?", id, user.ID)
+	result := db.DB.Raw("DELETE FROM apps"+
+		"INNER JOIN team_users ON team_users.team_id = apps.team_id"+
```
Comment (cjdenio) [inline pkg/api/apps/delete.go]: Use backticks instead of concatenation

**Review by cjdenio — CHANGES_REQUESTED** [review]
sorry, one more thing

File: pkg/api/apps/delete.go (hunk 3 above)
Comment (anirudhb) [reply pkg/api/apps/delete.go]: Don't want newlines

**Review by jasonappah — APPROVED** [review]
everything lgtm, merging

**Pull request merged**